src/sentry/static/sentry/app/components/group/pluginActions.jsx

```diff
@@ -0,0 +1,171 @@
+      },
+      error: error => {
+        addErrorMessage(t('Unable to unlink issue'));
+      },
+    });
+  },
+
+  loadPlugin(data) {
+    this.setState(
+      {
+        pluginLoading: true,
+      },
+      () => {
+        plugins.load(data, () => {
+          let issue = data.issue || null;
+          this.setState({pluginLoading: false, issue});
+        });
+      }
+    );
+  },
+
+  openModal() {
+    this.setState({
+      showModal: true,
+      actionType: 'create',
+    });
+  },
+
+  closeModal(data) {
+    this.setState({
+      issue: data.id && data.link ? {issue_id: data.id, url: data.link} : null,
+      showModal: false,
+    });
+  },
+
+  handleClick(evt) {
+    this.setState({
+      actionType: evt.target.id,
+    });
+  },
+
+  render() {
+    const {actionType, issue} = this.state;
+    const plugin = {...this.props.plugin, issue};
+
+    return (
+      <React.Fragment>
+        <IssueSyncListElement
+          onOpen={this.openModal}
+          externalIssueId={issue ? issue.issue_id : null}
+          externalIssueLink={issue ? issue.url : null}
+          onClose={this.deleteIssue}
+          integrationType={plugin.id}
+        />
+        <Modal
+          show={this.state.showModal}
+          onHide={this.closeModal}
+          animation={false}
+          backdrop="static"
+          enforceFocus={false}
+        >
+          <Modal.Header closeButton>
+            <Modal.Title>{`${plugin.name || plugin.title} Issue`}</Modal.Title>
+          </Modal.Header>
+          <ul
+            className="nav nav-tabs"
+            style={{borderBottom: '1px solid rgb(221, 221, 221)'}}
+          >
+            <li className={actionType == 'create' ? 'active' : ''}>
+              <a id="create" onClick={this.handleClick}>
+                Create
+              </a>
+            </li>
+            <li className={actionType == 'link' ? 'active' : ''}>
+              <a id="link" onClick={this.handleClick}>
+                Link
+              </a>
+            </li>
+          </ul>
+          {this.state.showModal && actionType == 'create' &&
+            !this.state.pluginLoading && (
+              <Modal.Body>
+                {plugins.get(plugin).renderGroupActions({
+                  plugin,
+                  group: this.getGroup(),
+                  project: this.getProject(),
+                  organization: this.getOrganization(),
+                  actionType: 'create',
+                  onSuccess: this.closeModal,
+                })}
+              </Modal.Body>
+            )}
+          {this.state.showModal && actionType == 'link' &&
+            !this.state.pluginLoading && (
+              <Modal.Body>
+                {plugins.get(plugin).renderGroupActions({
+                  plugin,
+                  group: this.getGroup(),
+                  project: this.getProject(),
+                  organization: this.getOrganization(),
+                  actionType: 'link',
```

Comment:
Similar to a previous comment, I think we don't need to have 2 separate branches based on `actionType`